src/main/java/com/google/sps/servlets/ListHabitServlet.java

```diff
@@ -0,0 +1,50 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.TimestampValue;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.sps.data.HabitList;
+
+
+// Servlet responsible for listing tasks. 
+@WebServlet("/list-habit")
+public class ListHabitServlet extends HttpServlet {
+    
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+     Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    
+    Query<Entity> query = Query.newEntityQueryBuilder().setKind("Habit").setOrderBy(OrderBy.desc("habitName")).build();
```

Comment:
I recommend with the repeated use of "habitName" and "Habit" to put it in a class constant.